src/frontend/src/handler/alter_table_column.rs

```diff
@@ -89,6 +96,15 @@ pub async fn replace_table_with_definition(
         table.columns.len(),
     );
 
+    let incoming_sink_ids: HashSet<_> = original_catalog.incoming_sinks.iter().copied().collect();
+    let target_columns = bind_sql_columns(&columns)?;
```

Comment:
Can we use the columns from the table catalog which computed in `generate_stream_graph_for_table`? In some case the columns are derived from its connectors.